### DISCUSSION TITLE
Include PVC object in backup but do not backup the data

### DISCUSSION THREAD

**Discussion by gorantornqvist** [vmware-tanzu/velero#4477] [Community support Q&A] [answered]
I tried the velero.io/exclude-from-backup=true on a PVC but that means that the PVC object isnt created at restore.
Is there a way to include the PVC object in the backup but exclude the data in the PVC?

PS: If there is, it should be included in the docs: https://velero.io/docs/v1.7/resource-filtering/

**Comment by sseago** [comment]
In my experience, if you're not defaulting to restic, velero will attempt to snapshot a PVC if it doesn't find a restic annotation. If it can't snapshot the volume, then it will restore the PVC without data. So there are probably 2 ways to accomplish this, depending on your volumes:
1) If snapshotting your volume isn't supported by a configured volume snapshotter plugin, then all you need to do is *not* set restic annotations.
2) If your volume does have snapshot support, then you'd need to not set restic annotations and make sure your backup isn't using a volume snapshot location.

I have more experience with 1) than 2) -- specifically, for NFS volumes, as long as you're not making restic backups/restores for your volumes, you should be able to restore the PVC/PV resources without actual data copy (of course, in the NFS case, you will still be pointing to the same NFS mount, so you will still have a volume with data on it, it just won't have been backed up or restored by velero+restic.

**Comment by gorantornqvist** [comment]
Hi,
Sorry for not being more specific, we are only using:
defaultVolumesToRestic: true
No snapshotting.

We are not setting any restic annotations anywhere.
Only annotations we set is backup.velero.io/backup-volumes: my-volume but thats for INCLUDING PVC/volumes since there seems to be some kind of bug in openshift-adp that we are using that causes volumes not be backed up if that annotation isnt there.

**Comment by sseago** [comment, marked as answer]
So if I'm understanding the use case correctly, you want some volumes to be backed up by restic and some to not be backed up at all (but with the PV/PVC resources backed up but empty), and you're not using snapshots at all? If I'm understanding correctly, there are two ways to do this:
1) if using `defaultVolumesToRestic: false`: set the restic annotation ("backup.velero.io/backup-volumes") for the volumes you want backed up on the appropriate pods and make sure that you don't have a VolumeSnapshotLocation configured for your backup (the issue with VSLs is that velero doesn't really have a notion of "I want the PV/PVC yaml but not the data" -- the only way to get this is for velero to think it wants to snapshot the PVC but it can't, so it just falls back to backing up and restoring the PV and PVC kube resources only)
2) if using `defaultVolumesToRestic: true` then it's just 1) in reverse. Set the annotation "backup.velero.io/backup-volumes-excludes" for the volumes that you *don't* want velero to back up with restic. Same as above -- you either need to have no configured VSL or your PVs need to be of a storage type that the VSL doesn't support.

**Reply by gorantornqvist** [reply]
Thanks! Both 1 and 2 worked fine!
